### DISCUSSION TITLE
Methods of parsing weather data in future weather stations

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#1248] [Ideas]
@bevans2000 @Urwumpe 

I find this [article](https://www.infoq.com/news/2024/01/1brc-fast-java-processing/) that talks about the contest of processing a billion row of temperature data interesting.

How should we be serializing and deserializing the weather data for each settlement when we transition from Java Serialization to JSON ? 

It'll become more important as we are on the path of creating new mission of building weather stations. 

Q1: can we come up with a more structural format for handling the weather data being collected ? 

Q2: should we save each param as a text files inside a default save folder for each default save ? 

Based on these data, I would love to create graphs and charts parsed from JSON or text files.

May be we can glean how teams from *One Billion Row Challenge (1BRC)*  would implement their [methods](https://github.com/gunnarmorling/1brc/) for mars-sim. 

```
The One Billion Row Challenge (1BRC) is a fun exploration of how far modern Java can be pushed for aggregating one billion rows from a text file. Grab all your (virtual) threads, reach out to SIMD, optimize your GC, or pull any other trick, and create the fastest implementation for solving this task!
```

**Comment by bevans2000** [comment]
I see any data stored in the data logger classes can be held externally outside the heap memory.
